library/std/src/sys/windows/net.rs

```diff
@@ -28,6 +28,10 @@ pub struct Socket(c::SOCKET);
 
 static INIT: Once = Once::new();
 
+// Workaround to prevent linking to `WS2_32.dll` when no network functionality is used.
+// See issue #85441.
+static mut CLEANUP: Option<unsafe extern "system" fn() -> i32> = None;
```

Comment:
Consider using `SyncLazy` for the `INIT`.

Reply:
I combined the two statics in a `SyncOnceCell`.